File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.py
from collections import deque
def solution(cacheSize, cities):
    answer = 0
    
    cities = deque(cities)
    q = []
    time = 0

    while cities:
        if cacheSize == 0:
            time += 5*len(cities)
            break
        c = cities.popleft()
        c = c.lower()
        if c not in q:
            if len(q) == cacheSize:
                q.pop(0)
            q.append(c)
            time += 5
        else:
            q.pop(q.index(c))
            q.append(c)
            time+=1
                
    
    return time

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;
class Solution {
    static int n;
    public int solution(int cacheSize, String[] cities) {
        int answer = 0;
        
        n = cities.length;
        Queue<String> q = new LinkedList<>();
        if(cacheSize == 0){
            return cities.length*5;
        } 
        
        for(String c: cities){
            c = c.toLowerCase();
            
            if(q.contains(c)){
                q.remove(c);
                q.offer(c);
                answer+=1;
            }else{
                if(q.size()==cacheSize){
                    q.poll();
                    q.offer(c);
                    answer+=5;
                }else{
                    q.offer(c);
                    answer+=5;
                }
            }
            
        }
        
        return answer;
    }
}